### ISSUE TITLE
The new password screen in incorrect

### ISSUE THREAD

**Issue by johanw666** [signalapp/Signal-Android#8106]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I use a password to unlock Signal and I didn't change it since the default functinality was changed to use the standard Android unlock mechanism (code, pattern or biometry). This still works but since 4.25.0 Signal seems to assume that "unlock" means "fingerprint scanner". I don't even have a fingerprint scanner on this device and if I had I would not use it for security reasons.

### Screenshots
![signal_20180814-164233](https://user-images.githubusercontent.com/8427572/44098973-15ef0414-9fd1-11e8-984c-291a31a7cc1c.jpg)

### Device info

**Device:** Sony Z3 Compact
**Android version:** 6.0.1
**Signal version:** 4.25.0

### Link to debug log
No debug log since it's a layout matter.

**Comment by art1fa** [comment]
Mh ok, so although this feature was basically removed, it somehow still works and with introduction of #8018 the layout is now broken.....does it really make sense to keep this legacy UI for a feature, we already replaced? 
Is there a specific reason you are not using the new screen lock feature which redirects to the Android system UI, @johanw666?

**Comment by johanw666** [comment]
@art1fa yes, I prefer it as an extra layer of protection against someone borrowing my device.

**Comment by greyson-signal** [comment]
@art1fa It's not just this variation. If you've set your system screenlock to be a pin or a password or whatever, you'll still see this screen right after entering your credentials in a system screen, and it'll incorrectly reference fingerprints then as well. 

In order to test this, remove fingerprint authentication from your device in your system settings, but keep a pint/password screen lock.

**Comment by leafcutterant** [comment]
> does it really make sense to keep this legacy UI for a feature, we already replaced?
> Is there a specific reason you are not using the new screen lock feature which redirects to the Android system UI, @johanw666?

@art1fa, yes it makes absolute sense to use Signal's own password mechanism, and not Andoid's screen lock.

That reason is **compartmentalization**.

I don't want to protect my Signal messages with the same mechanism I'm protecting my holiday photos with.
